File: cos/olympus/src/main/java/cos/olympus/game/Placeable.java
package cos.olympus.game;

public interface Placeable {
    int x();

    int y();

    static Placeable sample(int x, int y) {
        return new Placeable() {
            @Override public int x() {
                return x;
            }

            @Override public int y() {
                return y;
            }
        };
    }
}

File: cos/src/main/java/cos/olympus/game/Placeable.kt
package cos.olympus.game

interface Placeable {
    val x: Pos
    val y: Pos

    companion object {
        @JvmStatic
        fun sample(x: Pos, y: Pos): Placeable {
            return object : Placeable {
                override val x = x
                override val y = y
            }
        }
    }
}
